rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -544,8 +539,11 @@ public Pom download(GroupArtifactVersion gav,
                                 }
                             }
 
-                            // so that the repository path is the same regardless of username
-                            pom = pom.withRepository(MavenRepository.MAVEN_LOCAL_USER_NEUTRAL);
+                            String fullDefaultMavenLocalUri = MavenRepository.MAVEN_LOCAL_USER_NEUTRAL.getUri().replace("~", System.getProperty("user.home"));
+                            if (repo.getUri().equals(fullDefaultMavenLocalUri)) {
+                                // so that the repository path is the same regardless of username
+                                pom = pom.withRepository(MavenRepository.MAVEN_LOCAL_USER_NEUTRAL);
```

Comment:
Don't we serialize this information, and therefor might not want a machine-specific path?

Reply:
but anyway, it does not make sense to replace it if it's not actually the home one, right?